pages/api/generate-meals.ts

```diff
@@ -1,39 +1,55 @@
 import type { NextApiRequest, NextApiResponse } from "next";
+import { LangtailCompletion } from "@langtail/node";
 
-const generateMealIdeas = async (selectedIngredients: string[]) => {
-  const { LANGTAIL_API_KEY, LANGTAIL_API_URL } = process.env;
+const {
+  LANGTAIL_API_KEY,
+  LANGTAIL_WORKSPACE,
+  LANGTAIL_PROMPT,
+  LANGTAIL_ENVIRONMENT,
+  LANGTAIL_BASE_URL,
+  LANGTAIL_API_URL,
+} = process.env;
 
-  if (!LANGTAIL_API_KEY) {
-    throw new Error("LANGTAIL_API_KEY is not defined.");
-  }
+if (!LANGTAIL_API_KEY) {
+  throw new Error("LANGTAIL_API_KEY is not defined.");
+}
 
-  if (!LANGTAIL_API_URL) {
-    throw new Error("LANGTAIL_API_URL is not defined.");
+const generateMealIdeas = async (selectedIngredients: string[]) => {
+  if (!LANGTAIL_PROMPT) {
+    throw new Error("LANGTAIL_PROMPT is not defined.");
   }
 
-  const options = {
-    method: "POST",
-    headers: {
-      "X-API-Key": LANGTAIL_API_KEY,
-      "content-type": "application/json",
-    },
-    body: JSON.stringify({
-      stream: false,
-      variables: { selectedIngredients: JSON.stringify(selectedIngredients) },
-      doNotRecord: false,
-      messages: [{ role: "user", content: "Generate" }],
-    }),
-  };
-
-  const response = await fetch(LANGTAIL_API_URL, options);
-
-  if (!response.ok) {
-    const text = await response.text();
-    throw new Error(`Failed to generate meal ideas. ${text}`);
+  if (!LANGTAIL_ENVIRONMENT) {
+    throw new Error("LANGTAIL_ENVIRONMENT is not defined.");
   }
 
-  const data = await response.json();
-  return data.choices[0].message;
+  // Initialize the LangtailCompletion instance
+  const lt = new LangtailCompletion({
+    apiKey: LANGTAIL_API_KEY,
+    baseURL: LANGTAIL_BASE_URL || "https://api.langtail.com",
```

Comment:
```suggestion
```

I would remove this. Typical user would not need to change this.
In the future if we deploy langtail on premise we can add this to the docs. For now it would just add one unnecessary line I think

Reply:
@capaj does it default to api.langtail.com if it isn't set? if so, sure it makes sense to me.

Reply:
Yes it does